### DISCUSSION TITLE
Blazor Hybrid questions.

### DISCUSSION THREAD

**Discussion by alikoli** [oqtane/oqtane.framework#3676] [General] [answered]
I would like to get some information about the behavior of Oqtane regarding Blazor Hybrid. Normally when a new version of the module is published the framework compares dll checksums and download dlls when necessary. This is super cool. How does it work on mobile? In case of similar behavior, doesn't it violate this aap-store rule (taken from https://developer.apple.com/app-store/review/guidelines/).

2.5.2 Apps should be self-contained in their bundles, and may not read or write data outside the designated container area, nor may they download, install, or execute code which introduces or changes features or functionality of the app, including other apps. Educational apps designed to teach, develop, or allow students to test executable code may, in limited circumstances, download code provided that such code is not used for other purposes. Such apps must make the source code provided by the app completely viewable and editable by the user.

**Comment by sbwalker** [comment, marked as answer]
The Apple App Store has many restrictions... which seem to change constantly to support their business model. You may not be able to take advantage of Oqtane's dynamic servicing model on iOS due to the restrictions you mentioned above. However you could still create a mobile app - as long as you do not rely on the dynamic servicing model (ie. you would need to distribute your custom module assemblies with the .NET MAUI app). The dynamic servicing model in Oqtane is mainly being used in desktop environments - eliminating the need to update applications individually on every desktop.

**Comment by alikoli** [comment]
Distributing my custom model assembles with my MAUI app (in fact in IOS build) won't be a problem for me but I struggle to understand how to do it.
By default MAUI project has Oqtane.Client and Oqtane.Shared dlls loaded. I would add my modules, themes dlls, (and its dependancies). How should I specify that the framework knows that module dlls shouldn't be downloaded and are provided locally?

**Comment by markdav-is** [comment]
I'm in the process of packaging for iOS and have gained a fuller understanding of the module updating Oqtane is performing.  I personally believe there could be a way to embed a specific version of the required modules at build-time, but this will require an additional build system that uses calls Oqtane libraries to determine required assemblies for a site at build-time instead of run-time.  I have not taken the effort to make this yet, and it would require a new setting in Oqtane to ignore dynamic module updates.  So it's a big lift and might not be possible if this additional setting is not accepted into the framework.  For now, we are rolling out our product to a select group using testflight and have no plans for attempting a public store deployment.

**Comment by alikoli** [comment]
Thank you @markdav-is for your answer. We used to be in the Enterprise program with a private distribution so we could upload what we wanted. Unfortunatelly our apps are now targeted to customer employees, not our own. Because of this we need to use app store as our distribution source as our Enterprise membership was not prolongated by Apple. 

Actually, I don't need to ignore dynamic module updates. This just won't be used. Your answer made me look deeper a bit. As I understand for now. I can add all required assemblies to resources and during start time copy them to Path.Combine(FileSystem.Current.AppDataDirectory, "oqtane") folder. The rest should work as expected. I will give it a try.
